app/Models/Provider.php

```diff
@@ -16,9 +18,19 @@ class Provider extends Model
     public $incrementing = false;
     protected $primaryKey = "name";
     protected $keyType = "string";
+    protected $fillable = [
+        "name",
+        "url",
+        "color",
+    ];
 
-    public function cityProvider()
+    public static function query(): Builder
     {
-        return $this->belongsTo(CityProvider::class);
+        return parent::query();
+    }
```

Comment:
Do we need this?

Reply:
It's a change made in different commit, added when merging main to this branch. I do not use it, but it is used somewhere, for sure.